internal/core/src/segcore/SegmentGrowingImpl.cpp

```diff
@@ -102,7 +103,8 @@ SegmentGrowingImpl::Insert(int64_t reserved_offset,
         AssertInfo(field_id_to_offset.count(field_id),
                    fmt::format("can't find field {}", field_id.get()));
         auto data_offset = field_id_to_offset[field_id];
-        if (!indexing_record_.SyncDataWithIndex(field_id)) {
+        if (!indexing_record_.SyncDataWithIndex(field_id) ||
+            !indexing_record_.HasRawData(field_id)) {
```

Comment:
This may seem difficult to understand; please change it to insert_raw_to_index = indexing_record_.SyncDataWithIndex(field_id) && indexing_record_.HasRawData(field_id); if (!insert_raw_to_index) xxx , which means that only the index holds the original data and the index data has been matched with the inserted data, and no more data will be written to the chunk.

Reply:
replace `indexing_record_.SyncDataWithIndex(field_id) && indexing_record_.HasRawData(field_id)` with `indexing_record_.HasRawData(field_id)`, because `indexing_record_.HasRawData(field_id)` include `indexing_record_.SyncDataWithIndex(field_id)`.